### PR TITLE
Added chat page for Vendor and Admin

### PR DESCRIPTION
**Pull request by djanbolotov**
## Describe your changes

- Added new page for chats

Снимок экрана 2024-07-22 в 17 07 00

Снимок экрана 2024-07-22 в 16 27 21

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-253)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [ ] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/common/utils/formatDateForChat.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const milliseconds = 1000;
+  const daysInAYear = 365;
+  const oneDayDifference = 1;
+  const twoDayDifference = 2;
+  const millisecondsInADay = hours * seconds * milliseconds;
+  const diff = (now.getTime() - date.getTime()) / millisecondsInADay;
+
+  const timeFormat = 'p';
+  const monthAndDayFormat = 'MMMM d';
+  const yearMonthAndDayFormat = 'yyyy MM d';
```
Comment (lyudad) [inline src/common/utils/formatDateForChat.ts]: it should be constand date format through full project. please check general constant file

File: src/common/utils/formatDateForChat.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { format } from 'date-fns';
+import { t } from 'i18next';
+
+function formatDateForChatList(date: Date): string {
```
Comment (lyudad) [inline src/common/utils/formatDateForChat.ts:9]: I dont see test for this function, all utils should be covered

Comment (djanbolotov) [reply src/common/utils/formatDateForChat.ts:9]: Added tests for this function

File: src/common/utils/formatDateForChat.ts (hunk 1 above)
Comment (djanbolotov) [reply src/common/utils/formatDateForChat.ts]: Added this formats to constant file

**Pull request merged**